internal/testrunner/runners/system/servicedeployer/terraform.go

```diff
@@ -0,0 +1,89 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package servicedeployer
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/elastic-package/internal/compose"
+	"github.com/elastic/elastic-package/internal/files"
+	"github.com/elastic/elastic-package/internal/install"
+	"github.com/elastic/elastic-package/internal/logger"
+	"github.com/elastic/elastic-package/internal/stack"
+)
+
+// TerraformServiceDeployer is responsible for deploying infrastructure described with Terraform definitions.
+type TerraformServiceDeployer struct {
+	definitionsDir string
+}
+
+// NewTerraformServiceDeployer creates an instance of TerraformServiceDeployer.
+func NewTerraformServiceDeployer(definitionsDir string) (*TerraformServiceDeployer, error) {
+	return &TerraformServiceDeployer{
+		definitionsDir: definitionsDir,
+	}, nil
+}
+
+// SetUp method boots up the Docker Compose with Terraform executor and mounted .tf definitions.
+func (tsd TerraformServiceDeployer) SetUp(inCtxt ServiceContext) (DeployedService, error) {
+	logger.Debug("setting up service using Terraform service deployer")
+	terraformDeployerYml, err := install.ServiceDeployerComposeFile("terraform")
+	if err != nil {
+		return nil, errors.Wrap(err, "can't locate docker compose file for service deployer")
+	}
+
+	service := dockerComposeDeployedService{
+		ymlPath: terraformDeployerYml,
+		project: "elastic-package-service",
+	}
+	outCtxt := inCtxt
+
+	p, err := compose.NewProject(service.project, service.ymlPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not create docker compose project for service")
+	}
+
+	// Clean service logs
+	err = files.RemoveContent(outCtxt.Logs.Folder.Local)
+	if err != nil {
+		return nil, errors.Wrap(err, "removing service logs failed")
+	}
+
+	// Boot up service
+	tfEnvironment := tsd.buildTerraformExecutorEnvironment(inCtxt)
+	opts := compose.CommandOptions{
+		Env:       tfEnvironment,
+		ExtraArgs: []string{"--build", "-d"},
+	}
+	if err := p.Up(opts); err != nil {
+		return nil, errors.Wrap(err, "could not boot up service using docker compose")
+	}
+
+	// Build service container name
```

Comment:
If I'm following correctly, we're creating a container for the terraform executor here, not the service being tested, right? If so, could you please fix the comment and variable names?

Reply:
My bad, fixed. Thanks!